File: src/main/java/com/kpilszak/booksandstuffecommerce/dto/Purchase.java
package com.kpilszak.booksandstuffecommerce.dto;

import com.kpilszak.booksandstuffecommerce.entity.Address;
import com.kpilszak.booksandstuffecommerce.entity.Customer;
import com.kpilszak.booksandstuffecommerce.entity.Order;
import com.kpilszak.booksandstuffecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}

File: angular-books-and-stuff-ecommerce/src/app/common/purchase.ts
import { Customer } from "./customer";
import { Address } from "./address";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
  customer: Customer = null as unknown as Customer;
  shippingAddress: Address = null as unknown as Address;
  billingAddress: Address = null as unknown as Address;
  order: Order = null as unknown as Order;
  orderItems: OrderItem[] = [];
}
